skfda/preprocessing/dim_reduction/feature_extraction/_per_class_transformer.py

```diff
@@ -0,0 +1,219 @@
+
+        Check that it is scikit-learn-like and that
+        uses target data in fit.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        if not (
+            hasattr(self.transformer, "fit")  # noqa: WPS421
+            and hasattr(self.transformer, "transform")  # noqa: WPS421
+            and hasattr(self.transformer, "fit_transform")  # noqa: WPS421
+        ):
+            raise TypeError(
+                "Transformer should implement fit and "
+                "transform. " + str(self.transformer)
+                + " (type " + str(type(self.transformer)) + ")"
+                " doesn't",
+            )
+
+        tags = self.transformer._get_tags()  # noqa: WPS437
+
+        if tags['stateless'] or not tags['requires_y']:
+            warnings.warn(
+                f"Parameter ``transformer`` with type"  # noqa: WPS237
+                f" {type(self.transformer)} should use class information."
+                f" It should have the ``requires_y`` tag set to ``True`` and"
+                f" the ``stateless`` tag set to ``False``",
+            )
+
+    def fit(
+        self,
+        X: T,
+        y: np.ndarray,
+    ) -> PerClassTransformer:
+        """
+        Fit the model on each class.
+
+        It uses X as training data and y as target values.
+
+        Args:
+            X: FDataGrid with the training data.
+            y: Target values of shape = (n_samples).
+
+        Returns:
+            self
+        """
+        self._validate_transformer()
+        classes, class_feature_transformers = _fit_feature_transformer(
+            X,
+            y,
+            self.transformer,
+        )
+
+        self._classes = classes
+        self._class_feature_transformers_ = class_feature_transformers
+
+        return self
+
+    def transform(self, X: T) -> Union[DataFrame, np.ndarray]:
+        """
+        Transform the provided data using the already fitted transformer.
+
+        Args:
+            X: FDataGrid with the test samples.
+
+        Returns:
+            Eiter array of shape (n_samples, G) or a Data Frame
+            including the transformed data.
+        """
+        sklearn_check_is_fitted(self)
+
+        transformed_data = np.empty((len(X), 0))
+        for feature_transformer in self._class_feature_transformers_:
+            elem = feature_transformer.transform(X)
+            data = np.array(elem)
+            transformed_data = np.hstack((transformed_data, data))
+
+        if self.array_output:
+            for i in transformed_data:
+                if isinstance(i, (FDataGrid, FDataBasis)):
+                    raise TypeError(
+                        "There are transformed instances of FDataGrid or "
+                        "FDataBasis that can't be concatenated on a NumPy "
+                        "array.",
+                    )
+            return np.array(transformed_data)
+
+        return DataFrame(
```

Comment:
You should test also the `DataFrame` case, ideally with transformers returning `FData`.